src/w3d/math/quaternion.cpp

```diff
@@ -163,39 +170,125 @@ Quaternion Quaternion::Trackball(float x0, float y0, float x1, float y1, float s
 
 Quaternion Build_Quaternion(const Matrix3D &mat)
 {
-    static int _nxt[3] = { 1 , 2 , 0 };
-
+    static int _nxt[3] = {1, 2, 0};
     float tr, s;
     int i, j, k;
     Quaternion q;
+    tr = mat[0][0] + mat[1][1] + mat[2][2];
 
+    if (tr > 0.0f) {
+        s = GameMath::Sqrt(tr + 1.0);
+        q[3] = s * 0.5;
```

Comment:
Are we sure all these floating point literals are intended to be double? Doesn't look that way in Mac binary. It looks like the + 1.0 might be intentional though so I may need to look at implementing the double precision versions of the gamemath wrappers in the future.